File: backend/src/main/java/com/backend/model/Patient.java
package com.backend.model;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.List;

@Entity
@Table(name = "patients")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Patient {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "patient_id")
    private Long patientId;

    // Many patients belong to one owner
    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "owner_id", nullable = false)
    private Owner owner;

    @Column(name = "name", nullable = false, length = 100)
    private String name;

    @Column(name = "species", length = 100)
    private String species;

    @Column(name = "breed", length = 100)
    private String breed;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    @Column(name = "gender", length = 20)
    private String gender;

    @OneToMany(mappedBy = "patient", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Appointment> appointments;

    @OneToMany(mappedBy = "patient", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Record> records;

}

File: frontend-v1/types/patient.ts

// =======================
// Patient Request DTO
// =======================
export interface PatientRequest {
  ownerId: number;        // int64 -> number
  name: string;           // 0 - 100 characters
  species: string;
  breed: string;
  dateOfBirth: string;    // ISO date string (YYYY-MM-DD)
  gender: string;
}

// =======================
// Patient Response DTO
// =======================
export interface PatientResponse {
  patientId: number;      // int64 -> number
  ownerId: number;
  ownerName: string;
  name: string;
  species: string;
  breed: string;
  dateOfBirth: string;    // ISO date string
  gender: string;
}
